torch/quantization/fx/_equalize.py

```diff
@@ -59,18 +67,23 @@ def __init__(self, dtype=torch.quint8, qscheme=torch.per_tensor_affine,
                                                   factory_kwargs=factory_kwargs)
 
         self.equalization_scale = torch.empty(0)
+        self.equalization_shape: List[int] = []
 
     def forward(self, x_orig):
         if not (x_orig.ndim >= 2 and x_orig.ndim <= 5):
             raise ValueError("InputEqualizationObserver only supports Linear and Conv layers")
 
+        # Calculate the shape needed to reshape the equalization scale later (needed for Conv layers)
+        self.equalization_shape = [1] * x_orig.ndim
+        self.equalization_shape[1] = x_orig.size(1)
+
         return self.input_obs(x_orig)
 
     def get_input_minmax(self):
         return (self.input_obs.min_vals, self.input_obs.max_vals)
 
     def set_equalization_scale(self, equalization_scale):
-        self.equalization_scale = equalization_scale
+        self.equalization_scale = torch.reshape(equalization_scale, self.equalization_shape)
```

Comment:
if we ware doing reshaping when we use the scale, do we still need to do reshape here?

Reply:
We need to reshape here so that the input node can be multiplied correctly with the equalization_scale when going through the mul node in the graph. I think some of the reshapes in the scaling of the weights may be redundant, but I just kept them there just in case.